src/Compilers/CSharp/Portable/Binder/PatternExplainer.cs

```diff
@@ -0,0 +1,512 @@
+                    constraintType.Equals(evaluationType, TypeCompareKind.AllIgnoreOptions))
+                {
+                    var typedTemp = new BoundDagTemp(te.Syntax, te.Type, te);
+                    return SamplePatternForTemp(typedTemp, constraintMap, evaluationMap, requireExactType: true);
+                }
+
+                return null;
+            }
+
+            // Handle the special case of a null test and a type evaluation to unbox a nullable value type
+            string tryHandleUnboxNullableValueType()
+            {
+                if (evaluations.Length == 1 && constraints.Length == 1 &&
+                    constraints[0] is (BoundDagNonNullTest _, true) &&
+                    evaluations[0] is BoundDagTypeEvaluation { Type: var evaluationType } te &&
+                    input.Type.IsNullableType() && input.Type.StrippedType().Equals(evaluationType, TypeCompareKind.AllIgnoreOptions))
+                {
+                    var typedTemp = new BoundDagTemp(te.Syntax, te.Type, te);
+                    var result = SamplePatternForTemp(typedTemp, constraintMap, evaluationMap, requireExactType: false);
+                    // We need a null check. If not included in the result, add it.
+                    return (result == "_") ? "not null" : result;
+                }
+
+                return null;
+            }
+
+            // Handle the special case of a tuple pattern
+            string tryHandleTuplePattern()
+            {
+                if (input.Type.IsTupleType &&
+                    constraints.IsEmpty &&
+                    evaluations.All(e => e is BoundDagFieldEvaluation { Field: var field } && field.IsTupleElement()))
+                {
+                    var elements = input.Type.TupleElements;
+                    int cardinality = elements.Length;
+                    var subpatterns = new ArrayBuilder<string>(cardinality);
+                    subpatterns.AddMany("_", cardinality);
+                    foreach (BoundDagFieldEvaluation e in evaluations)
+                    {
+                        var elementTemp = new BoundDagTemp(e.Syntax, e.Field.Type, e);
+                        var index = e.Field.TupleElementIndex;
+                        if (index < 0 || index >= cardinality)
+                            continue;
+                        var oldPattern = subpatterns[index];
+                        var newPattern = SamplePatternForTemp(elementTemp, constraintMap, evaluationMap, requireExactType: false);
+                        subpatterns[index] = makeConjunct(oldPattern, newPattern);
+                    }
+
+                    return "(" + string.Join(", ", subpatterns) + ")" + (subpatterns.Count == 1 ? " { }" : null);
+                }
+
+                return null;
+
+                static string makeConjunct(string oldPattern, string newPattern) => (oldPattern, newPattern) switch
```

Comment:
>makeConjunct [](start = 30, length = 12)

Consider inlining in the one caller.

Reply:
I find this much clearer.

---
In reply to: [443076231](https://github.com/dotnet/roslyn/pull/44702#discussion_r443076231) [](ancestors = 443076231)